app/auth.py

```diff
@@ -55,3 +56,42 @@ def unauthorized_token():
                         'message': 'please send your authentication token'})
     response.status_code = 401
     return response
+
+
+def permission_required(permission):
+    """Route decorator to test user authorizations.
+
+    The decorator should be applied *after* the authentication decorator.
+    For example::
+
+        @api.route('/secure')
+        @auth.permission_required(Permission.ADMIN_USER)
+        @token_auth.login_required
+        def hello():
+            pass
+
+    Response with a 403 response if authorization fails. If the user is
+    not set (because the username/password was blank, or because a
+    token_required decorator was not applied, then a 401 response is sent.
+    Authorization requires authentication.
+    """
+    def decorator(f):
+        @wraps(f)
+        def decorated_function(*args, **kwargs):
+            if g.get('user', None) is None:
+                # user not authenticated
+                response = jsonify({'status': 401, 'error': 'unauthenticated',
```

Comment:
I think this is a standard pattern, see http://flask.pocoo.org/docs/0.10/patterns/apierrors/#registering-an-error-handler from the docs

[jsonify](http://flask.pocoo.org/docs/0.10/api/#flask.json.jsonify) makes a Response object and by default the status code is 200. The status in the JSON body is just a helper for the user; it's not the official HTTP status code.